patches/react-native+0.68.7.patch

```diff
@@ -1,5 +1,48 @@
+diff --git a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/devsupport/DevSupportManagerBase.java b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/devsupport/DevSupportManagerBase.java
+index f4d0e82..7fb0251 100644
+--- a/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/devsupport/DevSupportManagerBase.java
++++ b/node_modules/react-native/ReactAndroid/src/main/java/com/facebook/react/devsupport/DevSupportManagerBase.java
+@@ -20,6 +20,7 @@ import android.content.pm.PackageManager;
+ import android.graphics.Color;
+ import android.graphics.Typeface;
+ import android.hardware.SensorManager;
++import android.os.Build;
+ import android.util.Pair;
+ import android.view.Gravity;
+ import android.view.View;
+@@ -1107,7 +1108,7 @@ public abstract class DevSupportManagerBase implements DevSupportManager {
+       if (!mIsReceiverRegistered) {
+         IntentFilter filter = new IntentFilter();
+         filter.addAction(getReloadAppAction(mApplicationContext));
+-        mApplicationContext.registerReceiver(mReloadAppBroadcastReceiver, filter);
++        compatRegisterReceiver(mApplicationContext, mReloadAppBroadcastReceiver, filter, true);
+         mIsReceiverRegistered = true;
+       }
+ 
+@@ -1223,4 +1224,21 @@ public abstract class DevSupportManagerBase implements DevSupportManager {
+ 
+     return mSurfaceDelegateFactory.createSurfaceDelegate(moduleName);
+   }
++
++  /**
++   * Starting with Android 14, apps and services that target Android 14 and use context-registered
++   * receivers are required to specify a flag to indicate whether or not the receiver should be
++   * exported to all other apps on the device: either RECEIVER_EXPORTED or RECEIVER_NOT_EXPORTED
++   *
++   * <p>https://developer.android.com/about/versions/14/behavior-changes-14#runtime-receivers-exported
++   */
++  private void compatRegisterReceiver(
++      Context context, BroadcastReceiver receiver, IntentFilter filter, boolean exported) {
++    if (Build.VERSION.SDK_INT >= 34 && context.getApplicationInfo().targetSdkVersion >= 34) {
```

Comment:
Pretty sure this `targetSdkVersion` check shouldn't be here and is just cruft that makes the code more complicated to think about. But :shrug: React Native's code quality is rapidly becoming not our problem — and definitely the right thing here is to backport the commit verbatim, like this. (So long as it doesn't have bugs with direct behavior impact, anyway.)